generator/src/main/java/org/apache/kafka/message/checker/EvolutionVerifier.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.message.checker;
+
+import org.apache.kafka.message.MessageSpec;
+
+public class EvolutionVerifier {
+    private final MessageSpec topLevelMessage1;
+    private final MessageSpec topLevelMessage2;
+
+    public EvolutionVerifier(
+        MessageSpec topLevelMessage1,
+        MessageSpec topLevelMessage2
```

Comment:
Should we use names like "older" and "newer" here (or "initial" and "final")? It might make the code below more clear

Reply:
I've been using 1 and 2 throughout the code so changing this would be a big decision. I have to think about it...